FEXCore/Source/Interface/Core/Core.cpp

```diff
@@ -565,6 +565,16 @@ ContextImpl::GenerateIR(FEXCore::Core::InternalThreadState* Thread, uint64_t Gue
 
     for (size_t j = 0; j < CodeBlocks->size(); ++j) {
       const FEXCore::Frontend::Decoder::DecodedBlocks& Block = CodeBlocks->at(j);
+
+      bool BlockInForceTSOValidRange = false;
```

Comment:
There's some redundancy between this variable and the `!= end()` checks further below. We can either drop this variable or replace the iterator checks with it.

Reply:
They are both necessary - a memory access past the last volatile metadata entry but within a range will be a no-tso access. A memory access outside of the range will use the default setting